src/binding/kotlin_binding.cpp

```diff
@@ -37,7 +37,7 @@ bool KotlinBinding::refcount_decremented_unsafe() {
 }
 
 void KotlinBinding::set_kt_binding(jni::JObject j_object) {
-    JVM_CRASH_COND_MSG(kt_binding, "Trying to set a KtBinding on an already bound Object");
+    JVM_DEV_ASSERT(!kt_binding, "Trying to set a KtBinding on an already bound Object");
```

Comment:
What would this mean if this happens in a non dev build?

Reply:
Hard to say, this check is really just for debugging purpose in case we did something wrong when synchronizing bindings with their respective JVM instance.
Having this Assert triggers means that we either created 2 bindings for the same KtObject, or that the binding got set on the wrong native object and something is wrong in our memory manager.
It's not supposed to be anything the user can influence.